BusBoard.ConsoleApp/TflApi.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using RestSharp;
+using RestSharp.Authenticators;
+
+namespace BusBoard.ConsoleApp
+{
+    class TflApi
+    {
+        private const string BaseUrl = "https://api.tfl.gov.uk/";
+
+        readonly IRestClient client;
+
+        public TflApi()
+        {
+            client = new RestClient(BaseUrl);
+        }
+
+        public T Execute<T>(RestRequest request) where T : new()
+        {
+            var response = client.Execute<T>(request);
+
+            if (response.ErrorException != null)
+            {
+                const string message = "Error retrieving response.  Check inner details for more info.";
+                var tflException = new Exception(message, response.ErrorException);
+                throw tflException;
+            }
+            else if (response.StatusCode == HttpStatusCode.NotFound)
+            {
+                throw new ArgumentException("Stop was not found");
```

Comment:
I'm not sure would be the best use of `ArgumentException`. I'd suggest creating a custom exception, that way it should be more clear from the name what it's for 🙂

Reply:
Yeah definitely. Added in 17009b6 🙂